framework/core/hpp_buffer.cpp

```diff
@@ -0,0 +1,58 @@
+/* Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <core/hpp_buffer.h>
+
+namespace vkb
+{
+namespace core
+{
+HPPBuffer::HPPBuffer(vkb::core::HPPDevice &   device,
+                     vk::DeviceSize           size,
+                     vk::BufferUsageFlags     buffer_usage,
+                     VmaMemoryUsage           memory_usage,
+                     VmaAllocationCreateFlags flags) :
+    Buffer(device, static_cast<VkDeviceSize>(size), static_cast<VkBufferUsageFlags>(buffer_usage), memory_usage, flags)
+{}
+
+void HPPBuffer::flush() const
```

Comment:
hides the super class flush function and that function is not marked virtual.  (All functions in this file suffer from the same problem)

Reply:
HPPBuffer is derived protectively from vkb::core::Buffer. That is, technically, the functions in HPPBuffer do not hide those of vkb::core::Buffer, since they are not part of HPPBuffer's public interface.
To be more explicit, I could change that to private inheritance or even make vkb::core::Buffer a private member of HPPBuffer, making it an implementation-only detail.